packages/interaction-kit/src/application.ts

```diff
@@ -131,71 +134,54 @@ export default class Application {
 		return this.#components.get(customID);
 	}
 
-	getCommand(type: ApplicationCommandType, customID: string) {
-		return this.#commands.get(type).get(customID);
+	getCommand(type: ApplicationCommandType, name: string) {
+		return this.#commands.get(type).get(name);
 	}
 
-	// TODO: Should this be moved into Command?
-	async updateCommands() {
-		console.log("Checking for command updates in Development Server");
-
-		if (!process.env.DEVELOPMENT_SERVER_ID) {
-			throw new NoDevelopmentServerEnvironmentVariableError();
-		}
+	loadApplicationCommandDirectory(directoryPath: string) {
+		const directory = path.join(
+			path.dirname(url.fileURLToPath(import.meta.url)),
+			directoryPath
+		);
 
-		const guildID: Snowflake = process.env.DEVELOPMENT_SERVER_ID as Snowflake;
-		const json = await API.getGuildApplicationCommands(guildID);
+		fs.readdir(directory, async (err, files) => {
+			if (err) {
+				throw err;
+			}
 
-		// TODO: Handle errors
-		/**
-		 * Not in development server:
-		 *  { message: 'Missing Access', code: 50001 }
-		 */
+			for (const file of files) {
+				if (file.endsWith(".js")) {
+					const command = await import(path.join(directory, file));
+					this.addCommand(command.default);
+				}
+			}
+		});
 
-		const allCommands = Array.from(this.#commands.values())
-			.map((map) => Array.from(map.values()))
-			.flat();
+		return this;
+	}
 
-		for (const command of allCommands) {
-			const signature = json.find((cmd) => cmd.name === command.name);
+	loadMessageComponentDirectory(directoryPath: string) {
+		const directory = path.join(
+			path.dirname(url.fileURLToPath(import.meta.url)),
+			directoryPath
+		);
 
-			if (!signature) {
-				console.log(`\tCreating ${command.name}`);
+		fs.readdir(directory, async (err, files) => {
+			if (err) {
+				throw err;
+			}
 
-				try {
-					await API.postGuildApplicationCommand(guildID, command.serialize());
-				} catch (e: unknown) {
-					console.error(`\tProblem updating ${command.name}`);
-					console.error(e);
-				}
-			} else if (!command.equals(signature)) {
-				console.log(`\tUpdating ${command.name}`);
-
-				try {
-					await API.patchGuildApplicationCommand(guildID, {
-						...command.serialize(),
-						application_id: this.#applicationID,
-						id: signature.id,
-					});
-				} catch (e: unknown) {
-					console.error(`\tProblem updating ${command.name}`);
-					console.error(e);
+			for (const file of files) {
+				if (file.endsWith(".js")) {
```

Comment:
Same here; does this break TS projects?

Reply:
It shouldn't, because everything is compiled to JS before starting the application.

Reply:
ts-node can break

Reply:
I'm gonna punt on this for now and also just focus on supporting the JS template for now, and add TS support in a follow up